z3c/dependencychecker/modules.py

```diff
@@ -0,0 +1,398 @@
+            for filename in filenames:
+                if filename.endswith('.zcml'):
+                    yield cls(
+                        top_dir,
+                        os.path.join(path, filename),
+                    )
+
+    def scan(self):
+        tree = ElementTree.parse(self.path).getroot()
+
+        for element in self.ELEMENTS:
+            element_namespaced = self._build_namespaced_element(element)
+            for node in tree.iter(element_namespaced):
+                for attrib in self.ELEMENTS[element]:
+                    dotted_name = self._extract_dotted_name(node, attrib)
+                    if dotted_name:
+                        yield dotted_name
+
+    def _extract_dotted_name(self, node, attr):
+        if attr in node.keys():
+            dotted_name = node.get(attr)
+            if dotted_name.startswith('.'):
+                return
+
+            if dotted_name == '*':
+                return
+
+            return DottedName(
+                dotted_name,
+                file_path=self.path,
+                is_test=self.testing,
+            )
+
+    @staticmethod
+    def _build_namespaced_element(element):
+        if ':' in element:
+            namespace, name = element.split(':')
+            return '{{http://namespaces.zope.org/{0}}}{1}'.format(
+                namespace,
+                name,
+            )
+
+        return '{{http://namespaces.zope.org/zope}}{0}'.format(element)
+
+
+class FTIFile(BaseModule):
+
+    TYPES_FOLDER = '{0}types'.format(os.sep)
+
+    @classmethod
+    def search_files(cls, top_dir):
+        """Finds all XML files on the package"""
+        if top_dir.endswith('.py'):
+            return
+
+        for path, folders, filenames in os.walk(top_dir):
+            for filename in filenames:
+                if filename.endswith('.xml') and cls.TYPES_FOLDER in path:
+                    yield cls(
+                        top_dir,
+                        os.path.join(path, filename),
+                    )
+
+    def scan(self):
+        tree = ElementTree.parse(self.path).getroot()
+
+        for node in tree.iter('property'):
+            if 'name' in node.keys():
+                name = node.get('name')
+                if name == 'behaviors':
+                    for subnode in node.iter('element'):
+                        if 'value' in subnode.keys():
+                            yield DottedName(
+                                subnode.get('value'),
+                                file_path=self.path,
+                                is_test=self.testing,
+                            )
+                elif name in ('klass', 'schema'):
+                    if node.text:
+                        yield DottedName(
+                            node.text.strip(),
+                            file_path=self.path,
+                            is_test=self.testing,
+                        )
+
+
+class GSMetadata(BaseModule):
```

Comment:
Add a docstring (or comment) that this is about plone's generic setup stuff.
Someone from the django world will be puzzled, otherwise, when reading this code :-)

Reply:
Done